plugins/gradle/tooling-extension-api/src/org/jetbrains/plugins/gradle/model/ProjectImportAction.java

```diff
@@ -141,6 +148,13 @@ public void accept(@NotNull GradleBuild includedBuild) {
     if (isProjectsLoadedAction) {
       wrappedController.getModel(TurnOffDefaultTasks.class);
     }
+    myConverterExecutor.shutdown();
+    try {
+      myConverterExecutor.awaitTermination(Long.MAX_VALUE, TimeUnit.SECONDS);
```

Comment:
Waiting forever is the simplest strategy, but probably not the best :) What kind of handling would you prefer here?